datafusion/core/src/sql/planner.rs

```diff
@@ -638,7 +638,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             TableFactor::Table {
                 ref name, alias, ..
             } => {
-                let table_name = name.to_string();
+                // Trim quoted table name: "db_name" -> db_name
+                let table_name = name.to_string().trim_matches('\"').to_string();
```

Comment:
`trim_matches` also trims double, triple, etc. quotes. and doesn't care about symmetry (one start quote should be ended by one enrd quote).

I believe we should handle (to support quotes table name) the parsing at the sqlparser library:

https://github.com/sqlparser-rs/sqlparser-rs

Reply:
If we fix it there, I can handle bringing the new version of sqlparser into DataFusion, FYI

Reply:
Agree we should do this during SQL parsing, returning a unified pattern of table names with qualifiers.